src/procstat.c

```diff
@@ -1217,10 +1217,27 @@ static ssize_t procstat_fmt_u32_percentile(void *object, uint64_t arg, char *buf
 
 void procstat_histogram_u32_add_point(struct procstat_histogram_u32 *series, uint32_t value)
 {
-	unsigned reset;
+	unsigned reset, reset_interval_expired;
+	uint64_t reset_interval;
+
+	reset_interval = __atomic_load_n(&series->reset_interval, __ATOMIC_ACQUIRE);
+
+	struct timespec cur_time;
+	if (clock_gettime( CLOCK_REALTIME, &cur_time) == -1) {
+		return -1;
+	}
+
+	uint64_t time_since_last_reset = cur_time.tv_sec - series->last_reset_time;
+
+	if ((reset_interval > 0)&&(time_since_last_reset > reset_interval)) {
+		reset_interval_expired = 1;
```

Comment:
as previous comment no need for reset_interval_expired flag

Reply:
I aligned this to exiting style of this function. Similarly, 'reset' flag is not needed, but it makes the code easier to read.